src/mdoc/Verifier.ts

```diff
@@ -299,76 +313,156 @@ export class Verifier {
     }));
   }
 
+  /**
+   * Set the session transcript data to use for the verification.
+   *
+   * This is arbitrary and should match the session transcript as it will be calculated by the verifier.
+   * The transcript must be a CBOR encoded DataItem of an array, there is no further requirement.
+   *
+   * Example: `usingSessionTranscriptBytes(cborEncode(DataItem.fromData([a,b,c])))` where `a`, `b` and `c` can be anything including `null`.
+   *
+   * It is preferable to use {@link usingSessionTranscriptForOID4VP} or {@link usingSessionTranscriptForWebAPI} when possible.
+   *
+   * @param {Buffer} sessionTranscriptBytes - The sessionTranscriptBytes data to use in the session transcript.
+   * @returns {Verifier}
+   */
+  public usingSessionTranscriptBytes(sessionTranscriptBytes: Buffer): Verifier {
+    if (this.#sessionTranscriptBytes) {
+      throw new Error(
+        'A session transcript has already been set, either with .usingSessionTranscriptForOID4VP, .usingSessionTranscriptForWebAPI or .usingSessionTranscriptBytes',
+      );
+    }
+    this.#sessionTranscriptBytes = sessionTranscriptBytes;
+    return this;
+  }
+
+  /**
+   * Set the session transcript data to use for the verification as defined in ISO/IEC 18013-7 in Annex B (OID4VP), 2024 draft.
+   *
+   * This should match the session transcript as it will be calculated by the mdoc app.
+   *
+   * @param {string} mdocGeneratedNonce - A cryptographically random number with sufficient entropy.
```

Comment:
We are from the verifier side here, use a proper description

Reply:
👍🏼 see https://github.com/auth0-lab/mdl/pull/16/commits/10fe386b4b716b0d63d9e43291673657cbd2752f#diff-f2508718eb8e318b6393b4a87fa592e807459ec03ca673a133ae9634f9f5d09dR344
and https://github.com/auth0-lab/mdl/pull/16/commits/10fe386b4b716b0d63d9e43291673657cbd2752f#diff-f2508718eb8e318b6393b4a87fa592e807459ec03ca673a133ae9634f9f5d09dR347